apps/lib/s_socket.c

```diff
@@ -444,10 +450,9 @@ int do_server(int *accept_sock, const char *host, const char *port,
             i = (*cb)(asock, type, protocol, context);
         }
 
-        if (naccept != -1)
-            naccept--;
```

Comment:
Doesn't this break things for DTLS (i.e. where `type != SOCK_STREAM`)?

Reply:
Additionally, I'm not sure I see what the problem is here.  naccept from the man page indicates that it only allows the server to accept <naccept> connections before exiting.  The fact that the listener remains open doesn't imply that that the server will call SSL_accept on the listening socket again, so I'm not sure what the issue is here

Reply:
The server does not call SSL_accept, but since it is configured to listen with a backlog, it essentially accepts clients on the TCP level.

Moreover, when using `openssl s_server` as a utility, it cannot be run sequentially where each process only handles one client, as it keeps the socket open.

Reply:
> Doesn't this break things for DTLS (i.e. where `type != SOCK_STREAM`)?

That is a great question - might be a good idea to add a `type == SOCK_STREAM` check.

Reply:
@mattcaswell This won't break it as this logic is already inside a `type == SOCK_STREAM` check.

Reply:
I have updated my commit to include a change that addresses `type != SOCK_STREAM` cases.